monai/handlers/checkpoint_loader.py

```diff
@@ -92,15 +98,22 @@ def __call__(self, engine: Engine) -> None:
         """
         checkpoint = torch.load(self.load_path, map_location=self.map_location)
 
-        if not self.strict_shape:
-            k, _ = list(self.load_dict.items())[0]
-            # single object and checkpoint is directly a state_dict
-            if len(self.load_dict) == 1 and k not in checkpoint:
-                checkpoint = {k: checkpoint}
+        k, _ = list(self.load_dict.items())[0]
+        # single object and checkpoint is directly a state_dict
+        if len(self.load_dict) == 1 and k not in checkpoint:
+            checkpoint = {k: checkpoint}
 
-            # skip items that don't match data shape
+        # skip items that don't match data shape
```

Comment:
this comment says skipping the shape mismatches, but line 110 is actually checking the type, not the shape.  is there a test case or use case that `strict=True, strict_shape=False`?

Reply:
Hi @wyli ,

Thanks for your reminder, this is a previous comment line that I forgot to update.
Updated in the latest commit and also added check for `strict=True, strict_shape=False`.

Thanks.